### PR TITLE
feat: [PIE-3345]: Schema validation of env group yaml

### PR DESCRIPTION
**Pull request by prashantsharma04**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32033)

### CODE REVIEW

**Comment by prashantsharma04** [comment]
trigger messagecheck

File: 125-cd-nextgen/src/main/java/io/harness/cdng/envGroup/beans/EnvironmentGroupConfig.java [hunk 1]
```diff
@@ -34,13 +38,13 @@
 @JsonTypeInfo(use = NAME, include = WRAPPER_OBJECT)
 @JsonTypeName("environmentGroup")
 public class EnvironmentGroupConfig implements YamlDTO {
-  @EntityName String name;
-  @EntityIdentifier String identifier;
+  @NotNull @EntityName @Pattern(regexp = NGRegexValidatorConstants.NAME_PATTERN) String name;
+  @NotNull @EntityIdentifier @Pattern(regexp = NGRegexValidatorConstants.IDENTIFIER_PATTERN) String identifier;
 
   @NotNull @Trimmed String orgIdentifier;
```
Comment (namanvermah) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/envGroup/beans/EnvironmentGroupConfig.java:44]: If env group can be at org and project level, shouldn't these not null at org and project identifiers be removed?

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32033)* status: 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[120-ng-manager/src/main/java/io/harness/serializer/NextGenRegistrars.java, line 109 at r1](https://reviewable.io/reviews/harness/harness-core/32033#-MzsYpC07_4e4XNTVkHq:-MzsYpC17sQLgZiBlN0V:b-1j5phn) ([raw file](https://github.com/harness/harness-core/blob/afe7fede9529ed9c69e10f36ca8f3fe4f4e33e6c/120-ng-manager/src/main/java/io/harness/serializer/NextGenRegistrars.java#L109)):*
> ```Java
>                    .availableAtAccountLevel(true)
>                    .availableAtOrgLevel(true)
>                    .clazz(EnvironmentGroupConfig.class)
> ```

Wraper for EnvGroup needs to be there and not internal class



_Code quote:_
```Java
          .add(YamlSchemaRootClass.builder()
                   .entityType(EntityType.ENVIRONMENT_GROUP)
                   .availableAtProjectLevel(true)
                   .availableAtAccountLevel(true)
                   .availableAtOrgLevel(true)
                   .clazz(EnvironmentGroupConfig.class)
```

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32033#-:-Mzt2IBCCcA5io4XIj29:bnfp4nl)* status: 6 file(s) to review. 1 change requests. 1 of 2 approvals obtained.

**Review by prashantsharma04 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32033)* status: 6 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[120-ng-manager/src/main/java/io/harness/serializer/NextGenRegistrars.java, line 109 at r1](https://reviewable.io/reviews/harness/harness-core/32033#-MzsYpC07_4e4XNTVkHq:-Mzt4Jdm1vhkIIPCCXKN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/afe7fede9529ed9c69e10f36ca8f3fe4f4e33e6c/120-ng-manager/src/main/java/io/harness/serializer/NextGenRegistrars.java#L109)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Wraper for EnvGroup needs to be there and not internal class
</blockquote></details>

Done.
___
*[125-cd-nextgen/src/main/java/io/harness/cdng/envGroup/beans/EnvironmentGroupConfig.java, line 44 at r1](https://reviewable.io/reviews/harness/harness-core/32033#-MzsQLNCh3jv2IWEmBVM-r1-44:-Mzt4KDp0kAhqIl0i0pS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/afe7fede9529ed9c69e10f36ca8f3fe4f4e33e6c/125-cd-nextgen/src/main/java/io/harness/cdng/envGroup/beans/EnvironmentGroupConfig.java#L44)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

If env group can be at org and project level, shouldn't these not null at org and project identifiers be removed?
</blockquote></details>

Done.

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32033)* status: 1 change requests. 2. unresolved discussions.

**Review by archit-harness — APPROVED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32033)* status: 1. unresolved discussions.

**Review by namanvermah — APPROVED** [review]

**Review by namanvermah — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32033)* status: :shipit: complete! ✓

**Pull request merged**